api/net/SignalR/MessageHub.cs

```diff
@@ -14,19 +14,55 @@ public class MessageHub : Hub
 {
     #region Variables
     private readonly JsonSerializerOptions _serializerOptions;
+    private readonly ILogger _logger;
     #endregion
 
     #region Constructors
     /// <summary>
     /// Creates a new instance of a MessageHub, initializes with specified parameters.
     /// </summary>
     /// <param name="serializerOptions"></param>
-    public MessageHub(IOptions<JsonSerializerOptions> serializerOptions)
+    /// <param name="logger"></param>
+    public MessageHub(IOptions<JsonSerializerOptions> serializerOptions, ILogger<MessageHub> logger)
     {
         _serializerOptions = serializerOptions.Value;
+        _logger = logger;
     }
     #endregion
 
     #region Methods
+    /// <summary>
+    /// Allows a user to join a group for messages.
+    /// </summary>
+    /// <param name="name"></param>
+    /// <returns></returns>
+    [HubMethodName("JoinRoom")]
+    public async Task JoinRoomAsync(string name)
+    {
+        await this.Groups.AddToGroupAsync(this.Context.ConnectionId, name);
+    }
+
+    /// <summary>
+    /// Provides way for user to leave a group.
+    /// </summary>
+    /// <param name="name"></param>
+    /// <returns></returns>
+    [HubMethodName("LeaveRoom")]
+    public Task LeaveRoomAsync(string name)
+    {
+        return this.Groups.RemoveFromGroupAsync(this.Context.ConnectionId, name);
+    }
+
+    /// <summary>
+    /// Returns the message sent
+    /// </summary>
+    /// <param name="target"></param>
+    /// <param name="message"></param>
+    /// <returns></returns>
+    [HubMethodName("ping")]
+    public Task PingAsync(string target, object message)
```

Comment:
A nice simple way to test messages from the client.  You can send any type of message.